### DISCUSSION TITLE
Loading System.Security.Cryptography.RSA in fsi

### DISCUSSION THREAD

**Discussion by OnurGumus** [dotnet/fsharp#14524] [Q&A] [answered]
I would like to run the following code in fsi

```fsharp
let rsaKey = System.Security.Cryptography.RSA.Create()
```

But it complains with RSA is not defined.

According to docs here:
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsa.create?view=net-7.0
RSA is suppose to be inside System.Security.Cryptography.dll which is not part of nuget package.

What can I do?

**Comment by OnurGumus** [comment]
Ok using 
#r "System.Security.Cryptography" solved my problem

**Comment by baronfel** [comment, marked as answer]
You have to #r "System.Security.Cryptography";; in your FSI session to get access to the types and members from that DLL:

![image](https://user-images.githubusercontent.com/573979/210180722-19ffdf3b-532d-45fc-b9cb-0745024d3995.png)

I'm not sure this is _intended_, though - I thought for a program targeting net7.0 (which I thought FSI was suppose to simulate?) S.S.C would be part of the default set of opens. @vzarytovskii what _should_ the intended programming environment of FSI be?

**Comment by OnurGumus** [comment]
As an addition, I have searched my disk for where that file might be and one location was 
```
./.nuget/packages/microsoft.netcore.app.runtime.osx-x64/7.0.0/runtimes/osx-x64/lib/net7.0/System.Security.Cryptography.dll
```
Though my search isn't exhaustive.
